src/Component/src/Metadata/Resource/Factory/AttributesResourceMetadataCollectionFactory.php

```diff
@@ -134,6 +134,12 @@ private function getResourceWithDefaults(string $resourceClass, ResourceMetadata
             $resource = $resource->withName($resourceConfiguration->getName());
         }
 
+        if (null === $resource->getPluralName()) {
+            $resourcePluralName = $resourceConfiguration->getPluralName();
+
+            $resource = $resource->withPluralName($resourcePluralName);
+        }
```

Comment:
Can be optimized a little.

```suggestion
        if (null === $resource->getPluralName()) {
            $resource = $resource->withPluralName(
                $resourceConfiguration->getPluralName()
            );
        }
```